### ISSUE TITLE
Tag binding with google_tags_location_tag_binding on compute resources requires unique identifier (int64)

### ISSUE THREAD

**Issue by alexbacchin-asx** [hashicorp/terraform-provider-google#14210]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

In order to use the resource **google_tags_location_tag_binding** with compute resources (instance, disks, snapshots, etc) the binding needs to be done to the project number and  the resource unique identifier (int64) and not the name 

E.g that works!
```
 resource "google_tags_location_tag_binding" "binding" {
      + id        = (known after apply)
      + location  = "australia-southeast1-c"
      + name      = (known after apply)
      + parent    = "//compute.googleapis.com/projects/27050xxxxx/zones/australia-southeast1-c/disks/379577yyyyyyyy"
      + tag_value = "tagValues/28148078zzzzzz"
    }
```
At the moment only the **google_compute_instance** resource outputs the unique identifier via  _instance_id_ field

These improvements can be done in 2 options:
1) Output the unique identifier on taggable compute and data resources:
       - **google_compute_disk**
       - **google_compute_snapshot**
       - **google_compute_image**

2) Modify **google_tags_location_tag_binding** to fetch project number and compute resource unique identifier from compute API before tag binding. This would match the _gcloud resource-manager tags bindings create_

```
#### THIS DOES NOT WORK #####
 resource "google_tags_location_tag_binding" "binding" {
      + id        = (known after apply)
      + location  = "australia-southeast1-c"
      + name      = (known after apply)
      + parent    = "//compute.googleapis.com/projects/<project_id>/zones/australia-southeast1-c/disks/<disk_name>"
      + tag_value = "tagValues/28148078zzzzzz"
    }
```

### References

* #0000

**Comment by nphilbrook** [comment]
@rileykarson I'm curious why you tagged this with `upstream`? The compute API returns this value, it's just not in the attributes of the Terraform resource.

**Comment by rileykarson** [comment]
I thought I'd commented here at the time, sorry! The API has custom behaviour for this reference, and we would have expected it to just require a project id and resource name based on https://google.aip.dev/122#full-resource-names and https://google.aip.dev/cloud/2510#google-apis

**Comment by nphilbrook** [comment]
> I thought I'd commented here at the time, sorry! The API has custom behaviour for this reference, and we would have expected it to just require a project id and resource name based on https://google.aip.dev/122#full-resource-names and https://google.aip.dev/cloud/2510#google-apis

When you say "The API" you are referring to the CRM tags API, correct? I recently experienced significant pain trying to tag a compute disk (not even using Terraform, just exploring options with gcloud). I'm not sure if you have visibility to see [my support case](https://console.cloud.google.com/support/cases/detail/v2/44840242?project=wf-gcp-us-nphilbro9-prod) or not as a Google employee, but me and the support rep eventually got to the bottom of it - I was trying to use the disk name as the final element of the Full Resource Name and the CRM tags API requires the disk *ID*, which is not documented anywhere and the opposite of the naming expected by the asset IAM Policy Analyzer. It was incredibly confusing and infuriating when I tried to tag a disk, met with permission denied, and when passing the exact same Full Resource Name and permission to the policy analyzer, it said I had the permission.

Sorry for the rant, just an ongoing frustration with GCP's inability to get all services on board with one way to refer to objects.

**Comment by rileykarson** [comment]
> When you say "The API" you are referring to the CRM tags API, correct?

Yeah, the CRM Tags API. https://google.aip.dev/122#fields-representing-another-resource is clear "When a field represents another resource, the field should be of type string and accept the resource name of the other resource." (and since this can reference an arbitrary resource, https://google.aip.dev/122#full-resource-names means it should be a full resource name)

> Sorry for the rant, just an ongoing frustration with GCP's inability to get all services on board with one way to refer to objects.

Yeah, lack of consistency is a frustration I share.

**Comment by lineshthakr** [comment]
I am also getting same error

Error: Error creating LocationTagBinding: googleapi: Error 400: Request contains an invalid argument.
│ Details:
│ [
│   {
│     "@type": "type.googleapis.com/google.rpc.BadRequest",
│     "fieldViolations": [
│       {
│         "description": "Must be a valid One Platform resource name of a tag compatible regional resource",
│         "field": "binding.resource"
│       },
│       {
│         "description": "Resource type not supported in location europe-west3",
│         "field": "binding.resource"
│       }
│     ]
│   }
│ ]




Eg: 
# google_tags_location_tag_binding.binding will be created
  + resource "google_tags_location_tag_binding" "binding" {
      + id        = (known after apply)
      + location  = "europe-west3"
      + name      = (known after apply)
      + parent    = "projects/123450-----/regions/europe-west3/instances/6881441981107624783"
      + tag_value = "tagValues/2814813zzzzzzzz"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

**Comment by kumadee** [comment]
From provider version 5.15.0 onwards, the provider exposes the `disk_id` attribute. It allows to create the tag binding for a compute disk.

Refer https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/modules/compute-vm/tags.tf how to apply the tag bindings to the compute instance and disks.

References:
- https://github.com/hashicorp/terraform-provider-google/blob/main/CHANGELOG_v5.md#5150-feb-5-2024
- https://github.com/hashicorp/terraform-provider-google/pull/17112
